app/src/main/java/com/nextcloud/talk/controllers/CallController.java

```diff
@@ -1146,12 +1146,14 @@ public void onNext(SignalingSettingsOverall signalingSettingsOverall) {
                                 for (int i = 0; i < signalingSettingsOverall.getOcs().getSettings().getStunServers().size();
                                      i++) {
                                     iceServer = signalingSettingsOverall.getOcs().getSettings().getStunServers().get(i);
-                                    if (TextUtils.isEmpty(iceServer.getUsername()) || TextUtils.isEmpty(iceServer
-                                                                                                                .getCredential())) {
-                                        iceServers.add(new PeerConnection.IceServer(iceServer.getUrl()));
-                                    } else {
-                                        iceServers.add(new PeerConnection.IceServer(iceServer.getUrl(),
-                                                                                    iceServer.getUsername(), iceServer.getCredential()));
+                                    for (int j = 0; j < iceServer.getUrls().size(); j++) {
+                                        if (TextUtils.isEmpty(iceServer.getUsername()) || TextUtils.isEmpty(iceServer
```

Comment:
I'd probably flip the logic and check for username/credentials first else you do this in each iteration on the loop while the result is always the same. The downside is having two nearly identical loops.

Reply:
Actually stun never has credentials and turn always. Also we will use a v3 api for it, currently discussing in the server pr, so more changes incoming here